cpp/src/slice2cs/Gen.cpp

```diff
@@ -1972,24 +1972,30 @@ Slice::Gen::TypesVisitor::visitEnum(const EnumPtr& p)
     string ns = getNamespace(p);
     string scoped = fixId(p->scoped());
     EnumeratorList enumerators = p->enumerators();
-    const bool explicitValue = p->explicitValue();
+    const bool useSet = p->maxValue() - p->minValue() + 1 > static_cast<int64_t>(enumerators.size());
```

Comment:
How this `useSet` works? I was thinking that we should use the set when explicit values are used as explicit values can contain gaps and we only accept the explicit values

Reply:
I'll add a comment. 

Keep in mind that Slice enums do not allow 2 enumerators to have the same value. So if the distance between min and max is the number of enumerators, all the values between min and max are set, and we can use a simple range check during unmarshaling. For example:

```
enum Colors { Red = 1, White = 2, Blue = 3 }

enum ReversedColors { Blue = 3, White = 2, Red = 1 }
```